jerry-core/api/jerry-snapshot.c

```diff
@@ -91,7 +126,10 @@ snapshot_add_compiled_code (ecma_compiled_code_t *compiled_code_p, /**< compiled
     return 0;
   }
 
-  uint16_t start_offset = (uint16_t) (globals_p->snapshot_buffer_write_offset >> JMEM_ALIGNMENT_LOG);
+  /* The snapshot generator always parses a single file,
+   * so the base always starts right after the snapshot header. */
+  size_t buffer_offset = globals_p->snapshot_buffer_write_offset - sizeof (jerry_snapshot_header_t);
```

Comment:
Do we know for sure that `sizeof (jerry_snapshot_header_t)` is a multiple of `JMEM_ALIGNMENT_LOG`? Some lines above we assert on `globals_p->snapshot_buffer_write_offset` being properly aligned, but that ensured the correctness of the old code variant only.

Reply:
The alignment doesn't matter here, since substracting is a rounding down.